app/controllers/email_authorizations_controller.rb

```diff
@@ -0,0 +1,12 @@
+class EmailAuthorizationsController < ApplicationController
+  def verify
+    user = User.find_by(username: params[:username])
+    raise ActionController::RoutingError, "Not Found" unless user && current_user == user
+
+    email_authorization = user.email_authorizations.order("created_at DESC").first
```

Comment:
is there a chace that this will return nothing?

Reply:
It shouldn't because the controller action that sends the mailer also creates the email_authorization. An edge case probably exists though.